runtime/jit_vm/ctsupport.cpp

```diff
@@ -720,4 +720,76 @@ jitGetDeclaringClassOfROMField(J9VMThread *vmStruct, J9Class *clazz, J9ROMFieldS
 	return currentClass;
 }
 
+void
+jitAllMethodFromSignatureEndDo(J9MethodFromSignatureWalkState* state)
+{
+	J9JavaVM* vm = state->classLoaderWalkState.vm;
+	vm->internalVMFunctions->allClassLoadersEndDo(&state->classLoaderWalkState);
+}
+
+J9Method*
+jitAllMethodFromSignatureNextDo(J9MethodFromSignatureWalkState* state)
+{
+	UDATA result = 0;
+
+	J9JavaVM* vm = state->classLoaderWalkState.vm;
+	J9ClassLoader* classLoader = vm->internalVMFunctions->allClassLoadersNextDo(&state->classLoaderWalkState);
+	while (NULL != classLoader) {
+		J9Class* clazz = vm->internalVMFunctions->internalFindClassUTF8(state->vmThread, state->className, state->classNameLength, classLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+		if (NULL != clazz) {
+			J9JNINameAndSignature nameAndSig;
+			nameAndSig.name = reinterpret_cast<char*>(state->methodName);
+			nameAndSig.nameLength = state->methodNameLength;
+			nameAndSig.signature = reinterpret_cast<char*>(state->methodSig);
+			nameAndSig.signatureLength = state->methodSigLength;
+
+			result = vm->internalVMFunctions->javaLookupMethod(state->vmThread, clazz, reinterpret_cast<J9ROMNameAndSignature*>(&nameAndSig), NULL, J9_LOOK_JNI | J9_LOOK_NO_JAVA);
+			if (0 != result) {
+				break;
+			}
+		}
+
+		classLoader = vm->internalVMFunctions->allClassLoadersNextDo(&state->classLoaderWalkState);
+	}
+
+	return reinterpret_cast<J9Method*>(result);
+}
+
+J9Method*
+jitAllMethodFromSignatureStartDo(J9MethodFromSignatureWalkState* state, J9JavaVM* vm, UDATA flags, U_8* className, U_16 classNameLength, U_8* methodName, U_16 methodNameLength, U_8* methodSig, U_16 methodSigLength)
+{
+	state->className = className;
+	state->classNameLength = classNameLength;
+	state->methodName = methodName;
+	state->methodNameLength = methodNameLength;
+	state->methodSig = methodSig;
+	state->methodSigLength = methodSigLength;
+
+	J9VMThread *currentThread = vm->internalVMFunctions->currentVMThread(vm);
+	state->vmThread = currentThread;
+
+	UDATA result = 0;
+
+	J9ClassLoader* classLoader = vm->internalVMFunctions->allClassLoadersStartDo(&state->classLoaderWalkState, vm, J9CLASSLOADERWALK_INCLUDE_DEAD);
+	while (NULL != classLoader) {
+		J9Class* clazz = vm->internalVMFunctions->internalFindClassUTF8(currentThread, className, classNameLength, classLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+		if (NULL != clazz) {
+			J9JNINameAndSignature nameAndSig;
+			nameAndSig.name = reinterpret_cast<char*>(methodName);
+			nameAndSig.nameLength = methodNameLength;
+			nameAndSig.signature = reinterpret_cast<char*>(methodSig);
+			nameAndSig.signatureLength = methodSigLength;
```

Comment:
If you put `nameAndSig` in `J9MethodFromSignatureWalkState`, this copying could be eliminated.

Reply:
Fixed in 2b02a6f